okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java

```diff
@@ -0,0 +1,25 @@
+package org.folio.okapi.common;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Factory for WebClient instances to avoid web socket leaks.
+ */
+public final class WebClientFactory {
+  private static final Map<Vertx, WebClient> clients = new ConcurrentHashMap<>();
+
+  private WebClientFactory() {
+    throw new UnsupportedOperationException("Utility classes cannot be instantiated");
+  }
```

Comment:
Not tested. okapi-testing has a UtilityClassTester.

Reply:
Yes, thanks!